app/models/petition.rb

```diff
@@ -105,6 +107,23 @@ class Petition < ActiveRecord::Base
     where("signature_count >= ?", SystemSetting.value_of_key(SystemSetting::GET_AN_MP_SIGNATURE_COUNT).to_i)
   }
 
+  def sponsor_emails
+    @sponsor_emails || []
+  end
+
+  def sponsor_emails=(emails)
+    if emails.respond_to?(:each)
+      @sponsor_emails = emails
+    else
+      @sponsor_emails = parse_emails(emails)
+    end
+    @sponsor_emails = @sponsor_emails.select{ |e| Authlogic::Regex.email =~ e }.uniq.first(AppConfig.sponsor_count_max)
```

Comment:
Could we re-use ``Sponsor`` here rather than doing the validation "inline" and hoping that this method and Sponsor's validations remain in sync?